### PR TITLE
Recordings update

### PR DESCRIPTION
**Pull request by Gonzalo-Avalos-Ribas**
Changes to make the test pass after INT-11212

### CODE REVIEW

File: src/steps/resource-manager/batch/converters.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function createBatchApplicationEntity(
         _type: BatchEntities.BATCH_APPLICATION._type,
         _class: BatchEntities.BATCH_APPLICATION._class,
         id: data.id,
-        name: data.displayName,
+        name: data.displayName ?? data.name,
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/resource-manager/batch/converters.ts:73]: Check this property

File: src/steps/resource-manager/management-groups/index.ts [hunk 2]
```diff
@@ -90,7 +90,7 @@ export async function getGraphObjectsForManagementGroup(options: {
   );
 
   for (const child of managementGroup.children || []) {
-    if (child.type === '/providers/Microsoft.Management/managementGroups') {
+    if (child.type === 'Microsoft.Management/managementGroups') {
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/resource-manager/management-groups/index.ts:93]: change it to includes and/or revie if its possible to use a type from the sdk

File: src/steps/resource-manager/resources/index.test.ts [hunk 3]
```diff
@@ -308,6 +308,7 @@ describe('step - resource has resource lock relationships', () => {
       directory: __dirname,
       name: 'resource-manager-step-resource-has-resource-lock-relationships',
       options: {
+        recordFailedRequests: true,
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/resource-manager/resources/index.test.ts:311]: Add a comment why we need to fail a request

File: src/steps/resource-manager/storage/index.test.ts [hunk 4]
```diff
@@ -70,7 +70,7 @@ test('rm-storage-queues', async () => {
 
   const stepResults = await executeStepWithDependencies(stepTestConfig);
   expect(stepResults).toMatchStepMetadata(stepTestConfig);
-}, 20_000);
+}, 200_000);
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/resource-manager/storage/index.test.ts:73]: revert

File: test/helpers/recording.test.ts [hunk 5]
```diff
@@ -120,17 +123,17 @@ test('should not re-record azure resource-manager API calls that use an explicit
   const { keyVaults, diagnosticSettings } =
     await getKeyVaultsAndDiagnosticSettings(configFromEnv);
   await recording.stop();
-
+  await recording.flush();
```
Comment (Gonzalo-Avalos-Ribas) [inline test/helpers/recording.test.ts:126]: remove

File: test/helpers/recording.test.ts [hunk 6]
```diff
@@ -120,17 +123,17 @@ test('should not re-record azure resource-manager API calls that use an explicit
...
   const { keyVaults: badKeyVaults, diagnosticSettings: badDiagnosticSettings } =
     await getKeyVaultsAndDiagnosticSettings(mockInstanceConfig);
   await recording.stop();
 
-  expect(keyVaults).toEqual(badKeyVaults);
-  expect(diagnosticSettings).toEqual(badDiagnosticSettings);
-});
+  expect(keyVaults.length).toEqual(badKeyVaults.length);
+  expect(diagnosticSettings.length).toEqual(badDiagnosticSettings.length);
+}, 1000_000);
```
Comment (Gonzalo-Avalos-Ribas) [inline test/helpers/recording.test.ts:136]: revert the number

File: src/steps/resource-manager/subscriptions/client.ts [hunk 7]
```diff
@@ -71,13 +71,23 @@ export class J1SubscriptionClient extends Client {
         passSubscriptionId: false,
       },
     );
+    const d = new Date(Date.now());
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/resource-manager/subscriptions/client.ts:74]: move to a different function

File: src/steps/resource-manager/subscriptions/client.ts [hunk 8]
```diff
@@ -71,13 +71,23 @@ export class J1SubscriptionClient extends Client {
...
       // sendOperationRequest was the only way I found to change the API version with this sdk
       async () =>
         await serviceClient.sendOperationRequest(
-          { scope, options: { timeout: FIVE_MINUTES } },
+          {
+            scope,
+            options: {
+              timeout: FIVE_MINUTES,
+              filter: `properties/usageEnd eq '${new Date(
+                Date.now(),
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/resource-manager/subscriptions/client.ts:87]: check if date.now() is needed

**Review by VDubber — APPROVED** [review]

**Pull request merged**